### PR TITLE
Feature/pagination support

### PR DESCRIPTION
**Pull request by howhelan**
Add support for paginating multiple documents with the same `path`.  Passing a value for the `skip` query param indicates that there may be more than one document.  The optional `page_size` param can be used for limiting the number of results that are returned.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -113,16 +116,35 @@ mod filters {
...
-                let filter = bson::doc! { "path": query.path };
 
-                match db.collection("data").find_one(filter, filter_options).await {
-                    Ok(Some(doc)) => {
-                        let data: Data = bson::from_document(doc).unwrap();
-                        Ok(warp::reply::json(&data))
+                if let Some(_) = query.skip {
+                    // If there is a skip query param, retrieve collection
+
+                    let filter_options = mongodb::options::FindOptions::builder().skip(query.skip).limit(query.page_size).build();
```
Comment (ajpauwels) [inline src/main.rs:123]: Let's add a bounds check on page_size so people can't request absurd amounts of data

File: src/main.rs [hunk 2]
```diff
@@ -113,16 +116,35 @@ mod filters {
...
+
+                    match db.collection("data").find(filter, filter_options).await {
+                        Ok(mut cursor) => {
+                            let mut results_vector = Vec::new();
+                            while let Some(item) = cursor.next().await {
+                                results_vector.push(item.unwrap())
+                            }
+                            Ok(warp::reply::json(&results_vector))
+                        },
+                        Err(e) => Err(warp::reject::reject()),
```
Comment (ajpauwels) [inline src/main.rs:134]: Not for addressing in this PR, but we need to switch all these generic rejects into proper error-handling. I did the exact same thing in the FindOne route cuz I didn't know how to do proper warp error-handling, but let's make a separate PR that closes up these gaps.

**Review by ajpauwels — CHANGES_REQUESTED** [review]
Looks great, love the simplicity. One small bounds check to add and we'll address error-handling in a separate PR.

File: src/main.rs (hunk 1 above)
Comment (howhelan) [reply src/main.rs:136]: Added a limit of 100.  returns 400 of page_size if above limit.  

Also updated the Response object to have a `results` field which has the array of results, instead of just returning the array of results as the response.

**Review by ajpauwels — CHANGES_REQUESTED** [review]
Just switch and_then to map

**Review by ajpauwels — APPROVED** [review]
Not switching to map() since the filter returns a Result<>
